doc/source/structures/misc/lexicon.rst

```diff
@@ -0,0 +1,234 @@
+
+.. method:: Lexicon:REMOVE(key)
+
+    :parameter key: the keyvalue of the pair to be removed
+    
+    Remove the pair with the given key from the lexicon.
+    
+.. method:: Lexicon:CLEAR
+
+    Removes all of the pairs from the lexicon. Making it empty.
+    
+.. attribute:: Lexicon:LENGTH
+
+    :type: integer
+    :access: Get only
+
+    Returns the number of pairs in the lexicon.
+
+.. method:: Lexicon:COPY
+
+    :type: :struct:`Lexicon`
+    :access: Get only
+
+    Returns a new lexicon that contains the same set of pairs as this lexicon.
+
+.. method:: Lexicon:HASKEY(key)
+
+    :parameter key: (any type) 
+    :return: boolean
+
+    Returns true if the lexicon contains the provided key
+    
+.. method:: Lexicon:HASVALUE(key)
+
+    :parameter key: (any type) 
+    :return: boolean
+
+    Returns true if the lexicon contains the provided value
+    
+.. attribute:: Lexicon:DUMP
+
+    :type: string
+    :access: Get only
+
+    Returns a string containing a verbose dump of the lexicon's contents.
+
+.. attribute:: Lexicon:KEYS
+
+    :type: List
+    :access: Get only
+
+    Returns a string containing a verbose dump of the lexicon's contents.
+
+Access to Individual Elements
+-----------------------------
+
+``lexicon[expression]``
+    operator: another syntax to access the element at position 'expression'. Works for get or set. Any arbitrary complex expression may be used with this syntax, not just a number or variable name. 
+``FOR VAR IN LEXICON.KEYS { ... }.``
+    :ref:`A type of loop <flow>` in which var iterates over all the items of lexicon from item 0 to item LENGTH-1.
+
+Examples::
+
+    SET BAR TO LEXICON().       // Creates a new empty lexicon in BAR variable
+    BAR:ADD("FIRST",10).        // Adds a new element to the lexicon with the key of "FIRST"
+    BAR:ADD("SECOND",20).       // Adds a new element to the lexicon with the key of "SECOND"
+    BAR:ADD("LAST",30).         // Adds a new element to the lexicon with the key of "LAST"
+
+    PRINT BAR["FIRST"].            // Prints 10
+    PRINT BAR["SECOND"].            // Prints 20
+    PRINT BAR["LAST"].            // Prints 30
+
+    SET FOO TO LEXICON().           // Creates a new empty lexicon in FOO variable
+    FOO:ADD("ALTITUDE", ALTITUDE).  // Adds current altitude number to the lexicon
+    FOO:ADD("ETA", ETA:APOAPSIS).   // Adds current seconds to apoapsis to the lexicon at the index "ETA"
+
+    // As a reminder, at this point your lexicon, if you did all the above
+    // steps in order, would look like this now:
+    //
+    //  FOO["ALTITUDE"] = 99999. // or whatever your altitude was when you added it.
+    //  FOO["ETA:] = 99. // or whatever your ETA:APOAPSIS was when you added it.
+
+    PRINT FOO:LENGTH.        // Prints 2
+    PRINT FOO:LENGTH().      // Also prints 2.  LENGTH is a method that, because it takes zero arguments, can omit the parentheses.
+    SET x TO "ALTITUDE". PRINT FOO[x].  // Prints the same thing as FOO["ALTITUDE"].
+
```

Comment:
I assume you did a cut-n-paste from array docs and that's why everything from here on down looks a bit wonky is because you haven't got this far yet in re-editing what was cut-n-pasted?  Is that what you mean?